Implement docs feedback from heuristic walkthrough

cbethell opened the pull request:
**Issue Addressed**
NA

**What is the purpose of these changes?**
<!--Provide some background on the changes proposed.-->
This PR further incorporates the README feedback left on PR #221 and is therefore stacked on that PR's branch.

**What changes did you make?**
<!--Describe the concrete changes that you made or additional features that were added, be as detailed as possible in the steps you took.-->
I mostly incorporated the comments left on PR #221, with exception of providing link to example output files as suggested in https://github.com/AlexsLemonade/scpca-downstream-analyses/pull/221/files#r972402774, as we do not save the output files in the repo.

I also left some comments on that PR where I was unable to fully address a specific suggestion.

**Any comments, concerns, or questions important for reviewers**
- Does this PR appropriately address the comments left on PR #221?
- Should we maybe include screenshots of the output files since we cannot actually link to the output files?
- I was also not sure what was meant by adding some "possible version of the cmd" as suggested in https://github.com/AlexsLemonade/scpca-downstream-analyses/pull/221/files#r972388626.

**Checklist**

Place an `x` in all boxes that you have completed.

- [ ] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [x] I have added all necessary documentation (if applicable)

dvenprasad commented on README.md:
```diff
@@ -18,14 +18,14 @@ Cluster assignments are stored in the `SingleCellExperiment` object returned by
 <!-- cmd -->
 
-<!-- Output files -->
+There are two expected output files thay will be associated with each provided `SingleCellExperiment` object and `library_id`: a processed `SingleCellExperiment` object containing normalized data and clustering results, and a summary html report detailing the filtering of low quality cells, dimensionality reduction, and clustering that was performed within the workflow. See the [expected output s ...
```
```suggestion
**Output Files**
There are two expected output files thay will be associated with each provided `SingleCellExperiment` object and `library_id`: 
- A processed `SingleCellExperiment` object containing normalized data and clustering results
- A summary HTML report detailing:
    - Filtering of low quality cells
    - Dimensionality reduction
    - Clustering that was performed within the workflow. 

See the [expected output section](#expected-output) for more information on these output files.
```

Breaking it up so it is easier to scan. Please correct phrasing to fit this format, if needed.

dvenprasad commented on README.md:
```diff
@@ -181,11 +175,12 @@ The file should contain the following columns:
 ## Running the workflow
 
-We have provided a sankemake [configuration file](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html), `config.yaml` which sets the defaults for all parameters needed to run the workflow.
+We have provided a snakemake [configuration file](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html), [`config.yaml`](https://github.com/AlexsLemonade/scpca-downstream-analyses/blob/main/config.yaml) which sets the defaults for all parameters needed to run the workflow.
```
```suggestion
We have provided an example [snakemake configuration file](https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html), [`config.yaml`](https://github.com/AlexsLemonade/scpca-downstream-analyses/blob/main/config.yaml) which sets the defaults for all parameters needed to run the workflow.
```

dvenprasad commented on README.md:
```diff
@@ -18,14 +18,14 @@ Cluster assignments are stored in the `SingleCellExperiment` object returned by
-4. A local installation of Snakemake and either R or conda (see more on this in the ["how to install the core downstream analyses workflow" section](#how-to-install-the-core-downstream-analyses-workflow))
+1. A local installation of Snakemake and either R or conda (see more on this in the ["how to install the core downstream analyses workflow" section](#how-to-install-the-core-downstream-analyses-workflow))
+2. Single-cell gene expression data stored as `SingleCellExperiment` objects stored as RDS files (see more on this in the ["Input data format" section](#input-data-format))
+3. A project metadata tab-separated value (TSV) file containing relevant information about your data necessary for processing (see more on this in the ["Metadata file format" section](#metadata-file-format))
```
Maybe we can link the example metadata file here as well. Thinking about users who are coming back to the documentation after while. It would be nice for them to find what they might want without too much bouncing from section to section.

Could we also mention need to a config file here and link to the example file?

dvenprasad commented on README.md:
```diff
@@ -95,8 +78,10 @@ Therefore, you will also need to install Snakemake before running the pipeline.
 
-As described in the Snakemake instructions, the recommended way to install snakemake is using the conda package manager. <!-- do they need conda first? -->
-After installing [conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html) (we recommend the Miniconda installation), you can follow the steps below to set up the bioconda and conda-forge channels and install Snakemake in an isolated environment:
+As described in the Snakemake instructions, the recommended way to install snakemake is using the conda package manager. 
```
Can we make these two line more concise? Perhaps something like this:

```suggestion
Snakemake recommends installing it is using the conda package manager. 
```

dvenprasad commented on README.md:
```diff
@@ -95,8 +78,10 @@ Therefore, you will also need to install Snakemake before running the pipeline.
-As described in the Snakemake instructions, the recommended way to install snakemake is using the conda package manager. <!-- do they need conda first? -->
-After installing [conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html) (we recommend the Miniconda installation), you can follow the steps below to set up the bioconda and conda-forge channels and install Snakemake in an isolated environment:
+As described in the Snakemake instructions, the recommended way to install snakemake is using the conda package manager. 
+Therefore, you will want to first install [conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html) (we recommend the Miniconda installation).
```
```suggestion
Here are the instructions to install [conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html) 
We recommend the Miniconda installation.
```

dvenprasad reviewed (changes requested):
The changes are looking good! 

I also responded to [this comment](https://github.com/AlexsLemonade/scpca-downstream-analyses/pull/221#discussion_r981241892) on the other PR.  We can continue the discussion on this PR if that is easier. 

I left a few comments about minor formatting changes to make it easier to scan and read.

cbethell commented:
Thanks for the feedback @dvenprasad!! I added an example of that command you mentioned in https://github.com/AlexsLemonade/scpca-downstream-analyses/pull/221#discussion_r981241892 and _believe_ I implemented all of your suggestions here -- let me know what you think of things now!

dvenprasad commented on README.md:
```diff
@@ -95,8 +96,11 @@ Therefore, you will also need to install Snakemake before running the pipeline.
+Here are the instructions to [install conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html) 
+We recommend the Miniconda installation.
+
+After installing conda has been installed, you can follow the steps below to set up the bioconda and conda-forge channels and install Snakemake in an isolated environment:
```
Whoop rephrasing to smooth it out. 

```suggestion
After installing conda, you can follow the steps below to set up the bioconda and conda-forge channels and install Snakemake in an isolated environment:
```

dvenprasad reviewed (approved):
Just caught a little mismatched phrase. But other than that it looks good! Thank you!

🥭 🥭 🥭
